qiskit/algorithms/linear_solvers/matrices/linear_system_matrix.py

```diff
@@ -0,0 +1,34 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An abstract class for linear systems solvers in Qiskit's aqua module."""
+from abc import ABC, abstractmethod
+
+from qiskit import QuantumCircuit
+
+
+class LinearSystemMatrix(ABC):
```

Comment:
So this is expected to be implemented by a class that can access a circuit via self - the most obvious being a sub-class of QuantumCircuit where self is the circuit. Is it not feasible to have default implementation here if self is an instance of a QuantumCircuit?